ProjectSubmit/app/views/schools/index.html.erb

```diff
@@ -2,22 +2,34 @@
  
  <%= link_to 'Adicionar Escola', new_school_path %>
 <table>
-  <tr>
-    <th>Escola:</th>
 
-  </tr>
- 
+<%= form_tag(schools_path, :method => "get", id: "search-form") do %>
+  <%= text_field_tag :search, params[:search], placeholder: "Pesquisar Escolas" %>
+  <%= submit_tag "Search", :name => nil %>
+<% end %>
+
   <% @schools.each do |school| %>
     <tr>
+    <th>Escola:</th>
       <td><%= school.name %></td>
+      <th>Código:</th>
       <td><%= school.code %></td>
+      <th>Iniciais:</th>
       <td><%= school.initials %></td>
+      <th>Descrição:</th>
       <td><%= school.description %></td>
+      <th>Número de cursos existentes:</th>
+      <td><%= school.courses.size %></td>
+ 
+
 
       <td><%= link_to 'Mostrar', school_path(school) %></td>
       <td><%= link_to 'Editar', edit_school_path(school) %></td>
```

Comment:
Podes envolver os botões de editar / remover e até o de criar num bloco para que só sejam mostrados caso o utilizador tenha feito sign in.
```
<% if user_signed_in? %>
(...)
<% end %>
```